### PR TITLE
daf_persistence add functions to query if a storage exists at a location

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: python/lsst/daf/persistence/posixStorage.py [hunk 1]
```diff
@@ -601,6 +606,22 @@ def search(root, path, searchParents=False):
...
+        ----------
+        root : string
+            URI to the the root location of the storage
+
+        Returns
+        -------
+        bool
+            True if the storage exists, false if not
+        """
+        return os.path.exists(PosixStorage._pathFromURI(uri))
```
Comment (jhoblitt) [inline python/lsst/daf/persistence/posixStorage.py]: It looks like this boils down to a `stat(2)` call, so this should be nice and atomic.

**Comment by jhoblitt** [comment]
This looks reasonable to me but I am unfamiliar with this code.

File: python/lsst/daf/persistence/posixStorage.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def __init__(self, uri):
         :return:
         """
         self.log = Log.getLogger("daf.persistence.butler")
-        self.root = urllib.parse.urlparse(uri).path
+        self.root = self._pathFromURI(uri)
         if self.root and not os.path.exists(self.root):
             os.makedirs(self.root)
```
Comment (timj) [inline python/lsst/daf/persistence/posixStorage.py]: I think this should use the safe `safeMakeDir()`.

File: python/lsst/daf/persistence/posixStorage.py [hunk 3]
```diff
@@ -601,6 +606,22 @@ def search(root, path, searchParents=False):
             else:
                 return None
 
+    @staticmethod
+    def storageExists(uri):
```
Comment (timj) [inline python/lsst/daf/persistence/posixStorage.py]: This method is added but nothing in this commit seems to call it. Is it a specialization of a base implementation?

File: python/lsst/daf/persistence/posixStorage.py (hunk 2 above)
Comment (jhoblitt) [inline python/lsst/daf/persistence/posixStorage.py]: I missed this... is using the `exist_ok` param not enough?

File: python/lsst/daf/persistence/posixStorage.py (hunk 3 above)
Comment (n8pease) [inline python/lsst/daf/persistence/posixStorage.py]: it's called by the new code in storageContinued.py

File: python/lsst/daf/persistence/posixStorage.py (hunk 2 above)
Comment (jhoblitt) [inline python/lsst/daf/persistence/posixStorage.py]: I looked at `safeMakeDir` and I'm not sure if it is actually safe if the path it is given contains multiple [new] nested dirs.

File: python/lsst/daf/persistence/posixStorage.py (hunk 1 above)
Comment (jhoblitt) [inline python/lsst/daf/persistence/posixStorage.py]: I believe @timj's concern is valid.  Two threads could both make the `stat(2)` call while the dir does not exist and then collide on directory creation.

**Pull request merged**